agentarchives/atom/client.py

```diff
@@ -0,0 +1,648 @@
+        """
+        Fetches a list of resource URLs for every top-level description in the database.
+
+        :param string search_pattern: A search pattern to use in looking up resources by title or resourceid.
+            The search will match any title containing this string;
+            for example, "text" will match "this title has this text in it".
+            If omitted, then all resources will be fetched.
+        :param string identifier: Only records containing this identifier will be returned.
+            Substring matching will not be performed; however, wildcards are supported.
+            For example, searching "F1" will only return records with the identifier "F1", while searching "F*" will return "F1", "F2", etc.
+
+        :return: A list containing every matched resource's URL.
+        :rtype list:
+        """
+        skip = (page - 1) * page_size
+
+        params = {
+            'limit': page_size,
+            'skip': skip,
+            'topLod': '1',
+            'sf0': '_all',
+            'sq0': ''
+        }
+
+        if search_pattern:
+            params['sq0'] = '"' + self._escape_solr_query(search_pattern) + '"'
+
+        if identifier != '':
+            params['sf1'] = 'identifier'
+            params['sq1'] = self._escape_solr_query(identifier)
+
+        if sort_by is not None:
+            params['sort'] = 'alphabetic'
+            if sort_by == 'desc':
+                params['reverse'] = True
+
+        return self._get(self.api_base_url + '/informationobjects', params=params)
+
+    def count_collections(self, search_pattern='', identifier=''):
+        response = self._collections_search_request(search_pattern, identifier, 1)
+        return response.json()['total']
+
+    def find_collections(self, search_pattern='', identifier='', fetched=0, page=1, page_size=30, sort_by=None):
+        """
+        Fetches a list of all resource IDs for every resource in the database.
+
+        :param string search_pattern: A search pattern to use in looking up resources by title or resourceid.
+            The search will match any title or resourceid containing this string;
+            for example, "text" will match "this title has this text in it".
+            If omitted, then all resources will be fetched.
+        :param string identifier: Restrict records to only those with this identifier.
+            This refers to the human-assigned record identifier, not the automatically generated internal ID.
+            This value can contain wildcards.
+
+        :return: A list containing every matched resource's ID.
+        :rtype: list
+        """
+        def format_record(record):
+            # Get record details
+            full_record = self.get_record(record['slug'])
+            dates = self._fetch_dates_from_record(record)
+            date_expression = self._fetch_date_expression_from_record(record)
+
+            # Determine whether descendents exist
```

Comment:
It'll be slowish, but we can optimize later. I basically took the same approach as the Archivesspace implementation.